src/chainstate/stacks/miner.rs

```diff
@@ -1350,6 +1425,105 @@ impl StacksBlockBuilder {
         Ok(builder)
     }
 
+    /// This function defines the body of the `iterate_candidates` call in `build_anchored_block`.
+    /// It wraps StacksBlockBuilder::try_mine_tx_with_len that also skips processing according to
+    /// some pre-conditions and reacts to certain errors.
+    ///
+    /// We skip processing if:
+    ///  - the block limit for this block has been reached
+    ///  - if `txinfo.tx` is already in `considered`
+    ///  - if either of the origin or sponsor nonces is wrong
+    ///
+    ///  On `BlockTooBigError` or `TransactionTooBigError`, we increment the value of `block_limit_hit`.
+    fn process_candidate_for_anchored_block<'a>(
+        txinfo: MemPoolTxInfo,
+        epoch_tx: &mut ClarityTx<'a>,
+        builder: &mut StacksBlockBuilder,
+        considered: &mut HashSet<Txid>,
+        mined_origin_nonces: &mut HashMap<StacksAddress, u64>,
+        mined_sponsor_nonces: &mut HashMap<StacksAddress, u64>,
+        invalidated_txs: &mut Vec<Txid>,
+        block_limit_hit: &mut BlockLimitFunction,
```

Comment:
The fact that you have to pass so many `&mut` references to this method from the caller's scope suggests to me that this is not the right way factor this.